internal/testcli/golden.go

```diff
@@ -0,0 +1,220 @@
+	setKV(replacements, w.Config.ClusterID, "$DATABRICKS_CLUSTER_ID")
+	setKV(replacements, w.Config.WarehouseID, "$DATABRICKS_WAREHOUSE_ID")
+	setKV(replacements, w.Config.ServerlessComputeID, "$DATABRICKS_SERVERLESS_COMPUTE_ID")
+	setKV(replacements, w.Config.MetadataServiceURL, "$DATABRICKS_METADATA_SERVICE_URL")
+	setKV(replacements, w.Config.AccountID, "$DATABRICKS_ACCOUNT_ID")
+	setKV(replacements, w.Config.Token, "$DATABRICKS_TOKEN")
+	setKV(replacements, w.Config.Username, "$DATABRICKS_USERNAME")
+	setKV(replacements, w.Config.Password, "$DATABRICKS_PASSWORD")
+	setKV(replacements, w.Config.Profile, "$DATABRICKS_CONFIG_PROFILE")
+	setKV(replacements, w.Config.ConfigFile, "$DATABRICKS_CONFIG_FILE")
+	setKV(replacements, w.Config.GoogleServiceAccount, "$DATABRICKS_GOOGLE_SERVICE_ACCOUNT")
+	setKV(replacements, w.Config.GoogleCredentials, "$GOOGLE_CREDENTIALS")
+	setKV(replacements, w.Config.AzureResourceID, "$DATABRICKS_AZURE_RESOURCE_ID")
+	setKV(replacements, w.Config.AzureClientSecret, "$ARM_CLIENT_SECRET")
+	// setKV(replacements, w.Config.AzureClientID, "$ARM_CLIENT_ID")
+	setKV(replacements, w.Config.AzureClientID, "$USERNAME")
+	setKV(replacements, w.Config.AzureTenantID, "$ARM_TENANT_ID")
+	setKV(replacements, w.Config.ActionsIDTokenRequestURL, "$ACTIONS_ID_TOKEN_REQUEST_URL")
+	setKV(replacements, w.Config.ActionsIDTokenRequestToken, "$ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+	setKV(replacements, w.Config.AzureEnvironment, "$ARM_ENVIRONMENT")
+	setKV(replacements, w.Config.ClientID, "$DATABRICKS_CLIENT_ID")
+	setKV(replacements, w.Config.ClientSecret, "$DATABRICKS_CLIENT_SECRET")
+	setKV(replacements, w.Config.DatabricksCliPath, "$DATABRICKS_CLI_PATH")
+	setKV(replacements, w.Config.AuthType, "$DATABRICKS_AUTH_TYPE")
+}
+
+func PrepareReplacementsUser(t testutil.TestingT, replacements *orderedmap.OrderedMap[string, string], u iam.User) {
+	// There could be exact matches or overlap between different name fields, so sort them by length
+	// to ensure we match the largest one first and map them all to the same token
+	names := []string{
+		u.DisplayName,
+		u.UserName,
+		iamutil.GetShortUserName(&u),
+		u.Name.FamilyName,
+		u.Name.GivenName,
+	}
+	if u.Name != nil {
+		names = append(names, u.Name.FamilyName)
+		names = append(names, u.Name.GivenName)
+	}
+	for _, val := range u.Emails {
+		names = append(names, val.Value)
+	}
+	stableSortReverseLength(names)
+
+	for _, name := range names {
+		setKV(replacements, name, "$USERNAME")
+	}
+
+	for ind, val := range u.Groups {
+		setKV(replacements, val.Value, fmt.Sprintf("$USER.Groups[%d]", ind))
+	}
+
+	setKV(replacements, u.Id, "$USER.Id")
+
+	for ind, val := range u.Roles {
+		setKV(replacements, val.Value, fmt.Sprintf("$USER.Roles[%d]", ind))
+	}
+
+	// Schemas []UserSchema `json:"schemas,omitempty"`
+}
+
+func stableSortReverseLength(strs []string) {
+	slices.SortStableFunc(strs, func(a, b string) int {
+		return len(b) - len(a)
+	})
+}
+
+func NormalizeNewlines(input string) string {
```

Comment:
Did you find out what is causing the newline rewrites?

Reply:
No!